---
title: "LUSC survival curves"
output: html_document
date: "2025-05-22"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

My initial plan is to take TCGA-LUSC former smokers as a group, take a list of genes of interest ("persistent linked genes"), compare those former smokers with the highest and lowest 25% expression levels of the genes of interest, and compare survival times. My rationale: This could point to persistently genes that are involved in tumor progression and mortality in addition to carcinogenesis.

## Load libraries
```{r message=FALSE, warning=FALSE, results='hide'}
library(TCGAbiolinks)
library(SummarizedExperiment)
library(survival)
library(survminer)
library(dplyr)
```

## Read in the linked genes of interest
```{r}

LUSC_A1_TE_A2 <- read.table("../../2_Outputs/4_Linked_genes/LUSC_A1_TE_A2_20250514.txt")

```


## Download clinical data
```{r message=FALSE, warning=FALSE, results='hide'}

query_clinical <- GDCquery(
  project = "TCGA-LUSC",
  data.category = "Clinical",
  data.type = "Clinical Supplement"
)

# Check the available files
query_clinical$results[[1]] %>% head()

## I see that some are not BCR XML files, so I will try to remove these
query_clinical$results[[1]] <- query_clinical$results[[1]] %>%
  filter(data_format == "BCR XML")

## Download data
GDCdownload(query_clinical, directory = "../../GDC_data")
clinical_data <- GDCprepare_clinic(query_clinical, clinical.info = "patient", directory = "../../GDC_data")

head(clinical_data)

```


## Download expression data
```{r message=FALSE, warning=FALSE, results='hide'}
query <- GDCquery(project = "TCGA-LUSC",
                  data.category = "Transcriptome Profiling",
                  data.type = "Gene Expression Quantification",
                  sample.type = c("Primary Tumor", "Solid Tissue Normal"),
                  workflow.type = "STAR - Counts")

GDCdownload(query, directory = "../../GDC_data")

data <- GDCprepare(query, directory = "../../GDC_data")

counts <- as.data.frame(assay(data))  # Extracting the count matrix (these are supposedly raw counts)
head(counts)  # Viewing the first few rows (genes) and columns (samples)

gene_info <- as.data.frame(rowData(data))
head(gene_info)  # Preview the first few genes and their annotations

sample_info <- as.data.frame(colData(data))
head(sample_info)  # Preview sample metadata

```

## Survival analysis: Comparing former smokers (groups 3 and 4) with high and low expression of individual genes of interest 
```{r}
# Replace the sample barcodes with patient IDs (Note: maybe check more carefully that this worked properly later)
colnames(counts) <- sample_info$patient

# Subset data
former_smokers <- clinical_data[clinical_data$tobacco_smoking_history == c(3,4), ]
former_smokers <- former_smokers[!is.na(former_smokers$bcr_patient_barcode),] # Remove NA values

# Testing with particular genes of interest
genes_of_interest <- LUSC_A1_TE_A2$Gene

for (gene in genes_of_interest){
  
  gene_of_interest <- gene
  ID_of_interest <- gene_info[gene_info$gene_name == gene_of_interest,]$gene_id
  
  # Select the former smoker expression levels of this gene
  gene_of_interest_FS_expression_values <- counts %>%
    select(former_smokers$bcr_patient_barcode) %>%
    filter(rownames(.) %in% ID_of_interest)
    
  
  # Quantile thresholds
  low_threshold <- quantile(t(gene_of_interest_FS_expression_values), 0.25)
  high_threshold <- quantile(t(gene_of_interest_FS_expression_values), 0.75)
  
  # Group assignment
  former_smokers$group <- ifelse(t(gene_of_interest_FS_expression_values) <= low_threshold, "Low_25%",
                          ifelse(t(gene_of_interest_FS_expression_values) >= high_threshold, "High_25%", NA))
  
  former_smokers_high_low_goi <- former_smokers[!is.na(former_smokers$group),]
  
  # Survival object
  surv_object <- Surv(time = former_smokers_high_low_goi$days_to_death, 
                      event = !is.na(former_smokers_high_low_goi$days_to_death))
  
  # Kaplan-Meier fit
  fit <- survfit(surv_object ~ group, data = former_smokers_high_low_goi)
  
  # P-value
  p_value <- surv_pvalue(fit, data = former_smokers_high_low_goi)$pval
  
  if (p_value <= 0.05) {
    # Plot survival curves
    png(paste0("/Users/liambrockley/Desktop/Former_Smokers_Project/2_Outputs/", gene_of_interest, "_survival_plot.png"), width = 1600, height = 1200, res = 300) # Set up the plot to be printed
    ggsp <- ggsurvplot(fit, data = former_smokers_high_low_goi, pval = TRUE, title = paste0("LUSC Former smokers ", gene_of_interest))
    print(ggsp)
    dev.off()
  }
}

```

